Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -266,26 +313,66 @@ public final class PrometheusCollectorRegistry: Sendable {
             guard let entry = store[name] else {
                 // First time a Gauge is registered with this name.
                 let gauge = Gauge(name: name, labels: labels)
-                let gaugeWithHelp = MetricWithHelp(metric: gauge, help: help)
-                let gaugeGroup = MetricGroup(
-                    metricsByLabelSets: [key: gaugeWithHelp]
+                let gaugeFamily = MetricFamily(
+                    metricsByLabelSets: labels.isEmpty ? [:] : [key: gauge],
+                    help: help,
+                    metricUnlabeled: labels.isEmpty ? gauge : nil
                 )
-                store[name] = .gauge(gaugeGroup)
+                store[name] = .gauge(gaugeFamily)
                 return gauge
             }
+
             switch entry {
-            case .gauge(var existingGaugeGroup):
-                if let existingGaugeWithHelp = existingGaugeGroup.metricsByLabelSets[key] {
-                    return existingGaugeWithHelp.metric
+            case .gauge(var existingGaugeFamily):
+                // Check mutual exclusivity constraints. While Prometheus wouldn't break with sharing a
+                // metric name for labeled and unlabeled variants, the client enforces that each unique
+                // metric name must either be labeled (allowing multiple label sets) or unlabeled
+                // in order to support correct Prometheus aggregations.
+                if existingGaugeFamily.metricUnlabeled != nil && !labels.isEmpty {
```

Comment:
yeah if we centralize this series of checks in the Family type we don't have to duplicate this here, wdyt